CMakeLists.txt

```diff
@@ -42,9 +42,11 @@ if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
   set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "Debug" "Release" "MinSizeRel" "RelWithDebInfo")
 endif()
 
-# Default not to run the clang-tidy checks, default to whatever our CI_BUILD is
+# Default not to run the clang-tidy checks
```

Comment:
Why is this file changed?

Reply:
If we enable CI_BUILD without changing the file, then the builds will fail until all warnings are fixed and clang-tidy has been integrated with its warnings fixed. Rather than not using the `CI_BUILD` variable in a CI build, I have changed this file, noting that we should run clang-tidy and -Werror once those issues have been addressed properly. #431 will add clang-tidy to the CI_BUILD once the clang-tidy stuff has been fixed. We can do something similar for the warnings once #465 is merged.

Reply:
I dont think this is the right way to do this

Reply:
What do you propose?

Reply:
dont edit this file and get the other PRs merged in?